### PR TITLE
Spectral Analysis

### PR DESCRIPTION
**Pull request by Saurav-D**
# Peer Review Information

Added spectral analysis code

# Pull Request Check List

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: datasetinsights/stats/image_analysis/spectral_analysis.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    total_psd_1d = []
+    max_len = float("-inf")
+
+    for image in tqdm(images):
+        psd_2d = get_psd2d(image)
+        psd_1d = get_psd1d(psd_2d)
+        max_len = max(max_len, len(psd_1d))
+        total_psd_1d.append(psd_1d)
+
+    for i in range(len(total_psd_1d)):
```
Comment (davidwang-unity) [inline datasetinsights/stats/image_analysis/spectral_analysis.py:84]: we may use `np.vectorize` to apply functions to a list of numpy arrays for efficiency

Comment (Saurav-D) [reply datasetinsights/stats/image_analysis/spectral_analysis.py:84]: The [vectorize](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html#numpy.vectorize) function is provided primarily for convenience, not for performance. The implementation is essentially a for loop. Found this on the numpy docs.

Comment (davidwang-unity) [reply datasetinsights/stats/image_analysis/spectral_analysis.py:84]: good point! Thanks for pointing out that!

**Review by davidwang-unity — APPROVED** [review]
LGTM!

**Pull request merged**